### PR TITLE
Fix user list pagination

### PR DESCRIPTION
**Pull request by mpgxvii**
- Fix user list pagination by adding `metadata` property to `RestSourceUsers` to obtain page data and by fixing front end 

Solves #102

### CODE REVIEW

File: authorizer-app/src/app/components/rest-source-authorization/rest-source-user-list.component.html [hunk 1]
```diff
@@ -82,8 +82,10 @@
     </table>
 
     <mat-paginator
+      [length]="totalItems"
       [pageSize]="20"
-      [pageSizeOptions]="[5, 10, 20, 50]"
+      [pageSizeOptions]="[1, 5, 10, 20, 50]"
```
Comment (blootsvoets) [inline authorizer-app/src/app/components/rest-source-authorization/rest-source-user-list.component.html:87]: options 1 and 5 can be removed?

File: authorizer-app-backend/src/main/java/org/radarbase/authorizer/resources/RestSourceUserResource.kt [hunk 2]
```diff
@@ -64,7 +64,7 @@ class RestSourceUserResource(
         @QueryParam("source-type") sourceType: String?,
         @QueryParam("size") pageSize: Int?,
         @DefaultValue("1") @QueryParam("page") pageNumber: Int,
-    ): RestSourceUsers {
```
Comment (blootsvoets) [inline authorizer-app-backend/src/main/java/org/radarbase/authorizer/resources/RestSourceUserResource.kt:67]: Why not just add a `metadata` element to restsourceusers, including page, page size and total items? If we need metadata in another element, we can reuse the metadata type.

**Review by blootsvoets — COMMENTED** [review]
Looks OK, though I’d prefer putting the metadata in JSON to be explicit about it. In any case, using the `X-` header prefix is no longer recommended.

File: authorizer-app-backend/src/main/java/org/radarbase/authorizer/resources/RestSourceUserResource.kt [hunk 3]
```diff
@@ -74,7 +74,7 @@ class RestSourceUserResource(
                 .map { it.id }
         }
 
-        if (projects.isEmpty()) return RestSourceUsers(emptyList())
+        if (projects.isEmpty()) return RestSourceUsers(emptyList(), Page())
```
Comment (blootsvoets) [inline authorizer-app-backend/src/main/java/org/radarbase/authorizer/resources/RestSourceUserResource.kt:77]: for completeness, you can set
```suggestion
        if (projects.isEmpty()) return RestSourceUsers(emptyList(), Page(totalElements = 0))
```

**Review by blootsvoets — APPROVED** [review]
LGTM

**Comment by mpgxvii** [comment]
@blootsvoets Sorry would you be able to approve again? The latest commit dismissed the review.

**Review by blootsvoets — APPROVED** [review]

**Pull request merged**